### PR TITLE
Tenant qualifying URL support for JSP webapps

### PR DESCRIPTION
**Pull request by ashendes**
**Fixes:**
https://github.com/wso2/product-is/issues/8026
https://github.com/wso2/product-is/issues/8027

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 1]
```diff
@@ -171,46 +173,81 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
     }
 
     public String getAuthenticationEndpointURL() {
-        String authenticationEndpointURL = FileBasedConfigurationBuilder.getInstance().getAuthenticationEndpointURL();
-        if (StringUtils.isBlank(authenticationEndpointURL)) {
-            authenticationEndpointURL = "/authenticationendpoint/login.do";
+        String authenticationEndpointURL = "/authenticationendpoint/login.do";
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:176]: Shall we use a string constant for the url path here?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 2]
```diff
@@ -171,46 +173,81 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
...
     }
 
     public String getAuthenticationEndpointRetryURL() {
-        String authenticationEndpointRetryURL = FileBasedConfigurationBuilder.getInstance()
-                .getAuthenticationEndpointRetryURL();
-        if (StringUtils.isBlank(authenticationEndpointRetryURL)) {
-            authenticationEndpointRetryURL = "/authenticationendpoint/retry.do";
-        }
-        return authenticationEndpointRetryURL;
+            String authenticationEndpointRetryURL = "/authenticationendpoint/retry.do";
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:192]: Shall we use a string constant for the url?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 3]
```diff
@@ -171,46 +173,81 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
...
+
+            return authenticationEndpointRetryURL;
     }
 
     public String getAuthenticationEndpointWaitURL() {
-        String authenticationEndpointWaitURL = FileBasedConfigurationBuilder.getInstance()
-                .getAuthenticationEndpointWaitURL();
-        if (StringUtils.isBlank(authenticationEndpointWaitURL)) {
-            authenticationEndpointWaitURL = "/authenticationendpoint/wait.do";
+        String authenticationEndpointWaitURL = "/authenticationendpoint/wait.do";
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:207]: Shall we use a string constant for the url?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 4]
```diff
@@ -171,46 +173,81 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
...
+
         return authenticationEndpointWaitURL;
     }
 
     public String getIdentifierFirstConfirmationURL() {
-        String identifierFirstConfirmationURL = FileBasedConfigurationBuilder.getInstance()
-                .getIdentifierFirstConfirmationURL();
-        if (StringUtils.isBlank(identifierFirstConfirmationURL)) {
-            identifierFirstConfirmationURL = "/authenticationendpoint/idf-confirm.do";
+        String identifierFirstConfirmationURL = "/authenticationendpoint/idf-confirm.do";
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:223]: Shall we use string constants for the url paths?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 5]
```diff
@@ -171,46 +173,81 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
...
+
         return identifierFirstConfirmationURL;
     }
 
     public String getAuthenticationEndpointPromptURL() {
-        String authenticationEndpointPromptURL = FileBasedConfigurationBuilder.getInstance()
-                .getAuthenticationEndpointPromptURL();
-        if (StringUtils.isBlank(authenticationEndpointPromptURL)) {
-            authenticationEndpointPromptURL = "/authenticationendpoint/dynamic_prompt.do";
+        String authenticationEndpointPromptURL = "/authenticationendpoint/dynamic_prompt.do";
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:239]: Shall we fix the url path here too?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java [hunk 6]
```diff
@@ -219,14 +256,21 @@ public String getAuthenticationEndpointPromptURL() {
      * @return claims request URL
      */
     public String getAuthenticationEndpointMissingClaimsURL() {
-
-        String authenticationEndpointMissingClaimsURL = FileBasedConfigurationBuilder.getInstance()
-                .getAuthenticationEndpointMissingClaimsURL();
-        if (StringUtils.isBlank(authenticationEndpointMissingClaimsURL)) {
-            authenticationEndpointMissingClaimsURL = "/authenticationendpoint/claims.do";
+        String authenticationEndpointMissingClaimsURL = "/authenticationendpoint/claims.do";
```
Comment (ChamathNS) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:259]: Shall we use a string constant for the url?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 7]
```diff
@@ -440,11 +440,16 @@
...
+    "/authenticationendpoint/",
+    "/accountrecoveryendpoint/"
   ],
   "tenant_context.rewrite.servlets": [
     "/identity/(.*)",
-    "(.*)/identity/metadata/saml2"
+    "(.*)/identity/metadata/saml2",
+    "/commonauth(.*)",
+    "/logincontext(.*)",
+    "/oidc/"
```
Comment (ChamathNS) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:452]: Shall we put the changes of default.json and the identity.xml.j2 in a seperate PR? @malithie WDYT?

Comment (malithie) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:452]: yes please and link that one here

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java (hunk 1 above)
Comment (mefarazath) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java:176]: All the methods seem to have to the same logic,

1. Check tenant qualified and if so, build the tenant qualified URL
2. If not tenant qualified, build using the file-based config builder 

Just wondering whether we could extract the common login to a method that takes in functions as arguments

something like
buildUrl(String context, Function<String, String> legacyModeConfigFetchMethod)

**Comment by mefarazath** [comment]
Closing this PR since the changes introduced are merged with https://github.com/wso2/carbon-identity-framework/pull/2866

**Pull request closed without merging**